services-camel/src/test/java/edu/unc/lib/cdr/FulltextProcessorTest.java

```diff
@@ -0,0 +1,85 @@
+package edu.unc.lib.cdr;
+
+import static edu.unc.lib.cdr.headers.CdrFcrepoHeaders.CdrBinaryMimeType;
+import static edu.unc.lib.cdr.headers.CdrFcrepoHeaders.CdrBinaryPath;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_URI;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import edu.unc.lib.dl.fcrepo4.BinaryObject;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.Repository;
+import edu.unc.lib.dl.fedora.PID;
+
+public class FulltextProcessorTest {
+	private FulltextProcessor processor;
+	private final String slug = "full_text";
+	private final String fileSuffix = "-full_text.txt";
+	private final String testText = "Test text, see if it can be extracted.";
+	private File file;
+	private BinaryObject binary;
+	private FileObject parent;
+
+	@Mock
+	private Repository repository;
+	
+	@Mock
+	private Exchange exchange;
+	@Mock
+	private Message message;
+
+	@Before
+	public void init() throws Exception {
+		initMocks(this);
+		processor = new FulltextProcessor(this.repository, slug, fileSuffix);
+		file = File.createTempFile("testFile", "txt");
+		when(exchange.getIn()).thenReturn(message);
+		PIDs.setRepository(repository);
+		when(repository.getBaseUri()).thenReturn("http://fedora");
+	}
+	
+	@Test
+	public void validTest() throws Exception {
+		BufferedWriter writeFile = new BufferedWriter(new FileWriter(this.file));
+		writeFile.write(testText);
+		writeFile.close();
+		
+		String filePath = this.file.getAbsolutePath().toString();
+		
+		when(message.getHeader(eq(FCREPO_URI)))
+		.thenReturn("http://fedora/test/original_file");
+		
+		when(message.getHeader(eq(CdrBinaryPath)))
+				.thenReturn(filePath);
+		
+		when(message.getHeader(eq(CdrBinaryMimeType)))
+		.thenReturn("plain/text");
+		
+		binary = mock(BinaryObject.class);
+		parent = mock(FileObject.class);
+
+		when(repository.getBinary(any(PID.class))).thenReturn(binary);
+		when(binary.getParent()).thenReturn(parent);
+
+		processor.process(exchange);
+
+		verify(message).equals(FileObject.class);
```

Comment:
Take a look here:
https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/fcrepo4/deposit/src/test/java/edu/unc/lib/deposit/validate/ExtractTechnicalMetadataJobTest.java#L231

you can use an argument captor to get back the inputstream that was passed to the mock object, then read the string from the inputstream and make sure it matches the original test text.  For the string based parameters you can just use the eq(mimetype) type matcher.  The `verify` method is a wonderful thing